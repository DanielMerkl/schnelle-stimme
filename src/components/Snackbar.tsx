import React, { FC } from 'react';
import { IconButton, Snackbar as MuiSnackbar } from '@material-ui/core';
import { Close } from '@material-ui/icons';
import styled from 'styled-components';

interface Props {
  open: boolean;
  message: string;
  onClose: () => void;
}

export const Snackbar: FC<Props> = ({ open, message, onClose }) => {
  return (
    <MuiSnackbar
      anchorOrigin={{
        vertical: 'bottom',
        horizontal: 'left',
      }}
      open={open}
      onClose={onClose}
      message={message}
      autoHideDuration={5000}
      action={
        <StyledIconButton color="inherit" onClick={onClose}>
          <Close />
        </StyledIconButton>
      }
    />
  );
};

const StyledIconButton = styled(IconButton)`
  padding: 0.5rem;
`;
