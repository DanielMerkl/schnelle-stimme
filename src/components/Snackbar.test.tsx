import React from 'react';
import { render } from '@testing-library/react';
import { Snackbar } from './Snackbar';

describe('Snackbar', () => {
  it('renders correctly', () => {
    const { container } = render(
      <Snackbar open={true} message={'This is a banana.'} onClose={jest.fn()} />
    );

    expect(container).toMatchSnapshot();
  });
});
