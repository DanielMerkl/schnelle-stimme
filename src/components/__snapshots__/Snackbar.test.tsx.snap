// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snackbar renders correctly 1`] = `
<div>
  <div
    class="MuiSnackbar-root MuiSnackbar-anchorOriginBottomLeft"
  >
    <div
      class="MuiPaper-root MuiSnackbarContent-root MuiPaper-elevation6"
      direction="up"
      role="alert"
      style="opacity: 1; transform: scale(1, 1); transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,transform 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    >
      <div
        class="MuiSnackbarContent-message"
      >
        This is a banana.
      </div>
      <div
        class="MuiSnackbarContent-action"
      >
        <button
          class="MuiButtonBase-root MuiIconButton-root sc-AxjAm jQWxlQ MuiIconButton-colorInherit"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;
